db/migrations/20220222120300_remove_old_migrations_table.go

```diff
@@ -0,0 +1,54 @@
+package migrations
+
+import (
+	"time"
+
+	logging "github.com/RedHatInsights/sources-api-go/logger"
+	"github.com/go-gormigrate/gormigrate/v2"
+	"gorm.io/gorm"
+)
+
+// RemoveOldMigrationsTable removes the ActiveRecord migrations table.
+func RemoveOldMigrationsTable() *gormigrate.Migration {
+	type schemaMigrations struct {
+		Version string `gorm:"primaryKey;type:varchar"`
+	}
+
+	type arInternalMetadata struct {
+		Key       string    `gorm:"primaryKey;type:varchar;not null"`
+		Value     string    `gorm:"type:varchar"`
+		CreatedAt time.Time `gorm:"type:timestamp"`
+		UpdatedAt time.Time `gorm:"type:timestamp"`
+	}
+
+	return &gormigrate.Migration{
+		ID: "20220222120300",
+		Migrate: func(db *gorm.DB) error {
+			logging.Log.Info(`Migration "remove old migrations table" started`)
+			defer logging.Log.Info(`Migration "remove old migrations table" ended`)
+
+			err := db.Transaction(func(tx *gorm.DB) error {
+				err := tx.Migrator().DropTable(&schemaMigrations{})
+				if err != nil {
+					return err
+				}
+
+				err = tx.Migrator().DropTable(&arInternalMetadata{})
+				if err != nil {
+					return err
+				}
+
+				return nil
```

Comment:
I did have a thought here - can we add a `IF EXISTS` somehow? It might complain if the tables aren't there (e.g. local database lacks them)

Reply:
There is no need for that — the initial migration checks for the existence of that table, and if it doesn't exist, it creates it. So this migration will always find a table to delete, unless you run only this migration in isolation and not along with the rest.